scraper/src/merge_datasets.js

```diff
@@ -2,28 +2,95 @@ import fs from "fs";
 import { parse as json2csv } from "json2csv";
 import csv from "csvtojson";
 import permapeopleColumnMapping from "./helpers/column_mapping_permapeople.js";
+import axios from "axios";
+import axiosRetry from "axios-retry";
 import {
   sanitizeColumnNames,
   getSoilPH,
   getHeightEnumTyp,
   getSpreadEnumTyp,
-  fetchGermanName,
 } from "./helpers/helpers.js";
 
 /**
- * Fetches the German names for the plants from Wikidata API
+ * Defines the amount of retries we do, if axios encounters errors during a HTTP GET Request.
+ * Increse Delay if we encounter error to prevent 429 Errors.
  */
-const fetchGermanNames = async (plants) => {
-  return Promise.all(
-    plants.map(async (plant) => {
-      if (plant["common_name_de"]) {
-        return plant;
-      }
-      const germanName = await fetchGermanName(plant["unique_name"]);
-      plant["common_name_de"] = germanName;
-      return plant;
-    })
-  );
+axiosRetry(axios, {
+  retries: 5, // number of retries
+  retryDelay: (retryCount) => {
+    return retryCount * 1000; // time interval between retries
+  },
+});
+
+/**
+ * Fetches the German name of the plant from wikidata API.
+ * Sets the 'common_name_de of every plant in the array.
+ *
+ * @param {*} plants[]
+ */
+const fetchDataForPlantsArray = async (plants) => {
+  let GermanNamesFound = 0;
+  console.log("[INFO] Start fetching German common Names!");
+  for (const plant of plants) {
+    const unique_name = plant["unique_name"];
+    if (unique_name == "") {
+      continue;
+    }
+
+    try {
+      await axios
+        .get(
+          `https://www.wikidata.org/w/api.php?action=wbsearchentities&search=${unique_name}&language=en&format=json`
+        )
+        .then((response) => {
+          const data = response.data;
+          const results = data.search;
+
+          if (!results || results.length === 0) {
+            return null;
+          }
+
+          const result = results[0];
+          const id = result.id;
+
+          axios
+            .get(
+              `https://www.wikidata.org/w/api.php?action=wbgetentities&ids=${id}&languages=de&format=json`
+            )
+            .then((response) => {
+              const data = response.data;
+              const entities = data.entities;
+              const entity = entities[id];
+              const dewiki = entity["sitelinks"]["dewiki"];
+
+              if (dewiki) {
+                const title = dewiki.title;
+                let germanName = title.replace(/ \(.*\)/, "");
+                germanName = germanName.replace('"', "");
+
+                germanName = germanName.replace(unique_name, "");
+                germanName = germanName.trim();
+                if (!germanName || germanName === "true") {
+                  return null;
+                }
+                GermanNamesFound++;
+                plant["common_name_de"] = germanName;
```

Comment:
For "Allium sativum" it now says Knoblauchöl as German name? Should be Knoblauch.